update host.md (#31)
ZihengSun/Geoweaver: docs/host.md

## Read Python Environment

> 

Reviewer: How do people see the "Host Details" page?

Author: An additional step is added to mention how to get here.

> 1. Click the second icon from the left (python icon) at the bottom right corner of the "Host Details" page.

Reviewer: ```suggestion
1. Click the python icon button in the toolbar of the shown "Host Details" page.
```

> 2. Input the `Host User Password` of your account on the server in the shown dialog pop up.

Reviewer: What about the Localhost? Where do they find password for localhost?

Author: A note has been added at every occurrence of password input in all sections with a link to the password instructions in readme.md

> 3. An "Environment List" section will appear below the "Host Details" section showing all the available python environments on the connected server.

Reviewer: Are they really `all` available environments? Does it read both `pipenv` and `conda`? what about `virtualenv`?

Author: A "PyEnv" explicit callout has been added.

## File Browser

> 1. Click the second icon from the right (next to the red minus button) in the "Host Details" page.

Reviewer: Same problem. How do people access the `Host Details`.

Author: An additional step is added to mention how to get here.

> 1. Click the second icon from the right (next to the red minus button) in the "Host Details" page.

Reviewer: ```suggestion
1. Click the sitemap icon (the one with branches) button in the "Host Details" toolbar.
```

> 2. Input the `Host User Password` of your account on the server in the shown dialog pop up.

Reviewer: Same problem as above.

Author: An additional step is added to mention how to get here.

> 3. Click Confirm.

Reviewer: What happens after `confirm`? How do people know they do it right?

Author: Expanded the last step to mention the appropriate section appearing when the function executes correctly

## File Uploader

> 1. Click the upload file icon (the middle icon) at the bottom right corner of the "Host Details" page.

Reviewer: ```suggestion
1. Click the uploader icon button in the "Host Details" toolbar.
```

> 2. Input the `Host Password` of your account on the server in the shown dialog pop up.

Reviewer: same

Author: A note has been added at every occurrence of password input in all sections with a link to the password instructions in readme.md

> 3. A file upload section will appear below the "Host Details" section with drag and drop or file browser capability.

Reviewer: Better have a screenshot.

Author: A screenshot has been added and uploaded.

> 4. Choose the desired file to upload, and click the start button below the "File Uploader Section" to initiate the upload.

Reviewer: ```suggestion
4. Click on the `Open the File Browser` button to choose files, and click `Start` to initiate the uploading.
```

> 5. A progress bar with a status above will indicate the progress of the upload.

Reviewer: What is next? What happens when uploading finishes? Where did the file go? 

A tutorial should have beginning and end. Don't leave without giving people an end.

Author: Additional step is added to mention what happens when the upload is complete and where the file is.

## Edit a Host

1. Click the edit icon (first icon) at the bottom right corner of the "Host Details" page.

> 2. Change the desired fields.

Reviewer: ```suggestion
2. Change the field values as you wish. 
Note: The Host Id is not editable. 
```

> 3. Click the same edit icon as the first step. The changes have now been updated.

Reviewer: No need to click the edit button every time. Once you finish, click any place outside the text fields will trigger the save action. It is listening to the focus event.

Author: This doesn't work with me. if i click outside any of the fields, or even the right panel, nothing happens. only when the edit icon is clicked is when the values are actually updated.

Reviewer: That is how the fields in process/workflow pages work. It is fine. Just describe how it works in Host page.

## Read Python Environment

1. Select the desired host under `Hosts` on the right panel.

2. Click the python icon button in the toolbar of the shown "Host Details" page.

2. Input the `Host User Password` of your account on the remote server in the shown dialog pop up.

Note: if the selected server is localhost, password instructions are discussed [here](https://github.com/ESIPFed/Geoweaver#reset-password-for-localhost "Reset Password for Localhost")

> 3. An "Environment List" section will appear below the "Host Details" section showing all the available PyEnv python environments on the connected server.

Reviewer: ```suggestion
3. An `Environment List` section will appear below the `Host Details` section showing all the available PyEnv python environments on the connected server.
```

3. Click Confirm. A `File Browser Section` will appear below showing all the available directories and files.

## File Uploader

1. Select the desired host under `Hosts` on the right panel.

2. Click the uploader icon button in the "Host Details" toolbar.

2. Input the `Host Password` of your account on the server in the shown dialog pop up.

Note: if the selected server is localhost, password instructions are discussed [here](https://github.com/ESIPFed/Geoweaver#reset-password-for-localhost "Reset Password for Localhost")

3. A file upload section will appear below the "Host Details" section with drag and drop or file browser capability.
![Screen Shot 2022-01-15 at 12 39 12 AM](https://user-images.githubusercontent.com/34227630/149588755-f1982927-a2b0-453f-8248-65fc5668428c.png)

4. Click on the `Open the File Browser` button to choose files, and click `Start` to initiate the uploading.

5. A progress bar with a status above will indicate the progress of the upload.

6. The status with the message "Upload Complete" in green will show when the file is successfully uploaded, and the file is uploaded to the home directory of the logged in user.

## Edit a Host

> 1. Click the edit icon (first icon) at the bottom right corner of the "Host Details" page.

Reviewer: ```suggestion
1. Click the edit icon (first icon) in the `Host Details` page.
```

2. Change the field values as you wish. 
Note: The Host Id is not editable. 

> 3. Click the same edit icon as the first step. The changes have now been updated.

Reviewer: ```suggestion
3. Click the edit icon button again. The changes should be updated.
```
